### PR TITLE
Console cleanup

### PR DESCRIPTION
**Pull request by k-doering-NOAA**
## What issue(s) does this PR address? Describe and add issue numbers, if applicable.

Link issue(s) here: 

#313 . The new ADMB console (#257) inspired me to make some changes to the SS3 console output

## What tests have been done? Upload any model input files created for testing in a zip file, if possible.

Some of the changes have been tested locally by compiling and running a model

## What tests/review still need to be done? Who can do it, and by when is it needed (ideally)?

When the last changes are made, the output should be looked at in github actions to make sure everything looks ok

## Check which is true. This PR requires:

- [ ] no further changes to r4ss
- [x] no further changes to the manual
- [ ] no further changes to SSI (the SS3 GUI)
- [ ] no further changes to the stock synthesis change log (new features, bug reports)

## Describe any changes in r4ss/SS3 manual/SSI/change log that are needed (if not checked):

May need to check on r4ss and SSI

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
For reference, this is what the admb console output looks like for the simple admb model:
```
Starting optimization of 'simple' in phase 1 of 1 at Thu May 12 12:33:34 2022
phase= 1 | nvar=  2 | iter=  0 | nll=2.50e+01 | mag=3.61e+00 | par[  2]=b1
phase= 1 | nvar=  2 | iter=  7 | nll=3.45e+00 | mag=7.00e-05 | par[  2]=b1
Optimization completed after 0 s with final statistics:
 nll=3.451260 | mag=7.00140e-05 | par[  2]=b1

Checking for estimated parameters on bounds... none found!
Calculating Hessian: 1, 2 done! (0.019 s)
Inverting Hessian: 1, 2 done! (0 s)
Starting standard error calculations...  done! (0 s)

Finished running model 'simple' after 0.019 s.
```

This is what the SS3 console output looks like (including current changes made to this branch):

```
Begin reading the 4 required input files
Finished reading the 4 required input files

Begin setting up parameters
Finished setting up parameters
Begin evaluating biology calculations once
Finished evaluating biology calculations once


Begin estimating
Starting optimization of 'ss' in phase 1 of 5 at Thu May 12 12:30:50 2022
 did survey obj_fun  0 485.216 213.444
 did lencomp obj_fun   1613.44 408.812 0
 did agecomp obj_fun  2067.9 570.293 0
 did meanlength obj_fun  0 0 0
 initequ_catch -log(L)  0 0 0
 catch -log(L)  140920 0 0
 did recruitdev obj_fun -15.8356 31 0.72
 OK with obj_func 144812
phase= 1 | nvar=  3 | iter=  0 | nll=1.45e+05 | mag=1.98e+07 | par[  1]=SR_parm[1]
phase= 1 | nvar=  3 | iter= 20 | nll=9.09e+02 | mag=2.74e+02 | par[  1]=SR_parm[1]
phase= 1 | nvar=  3 | iter= 23 | nll=9.08e+02 | mag=8.54e+00 | par[  1]=SR_parm[1]
phase= 2 | nvar= 38 | iter=  0 | nll=9.08e+02 | mag=3.47e+02 | par[ 25]=recdev1[23]
phase= 2 | nvar= 38 | iter= 20 | nll=8.24e+02 | mag=5.65e+03 | par[  2]=SR_parm[1]
phase= 2 | nvar= 38 | iter= 40 | nll=7.99e+02 | mag=1.00e+03 | par[  2]=SR_parm[1]
phase= 2 | nvar= 38 | iter= 56 | nll=7.96e+02 | mag=1.29e+00 | par[ 18]=recdev1[16]
phase= 3 | nvar= 40 | iter=  0 | nll=7.96e+02 | mag=3.78e+00 | par[ 39]=selparm[4]
phase= 3 | nvar= 40 | iter= 20 | nll=7.96e+02 | mag=2.55e+01 | par[  2]=SR_parm[1]
phase= 3 | nvar= 40 | iter= 35 | nll=7.96e+02 | mag=9.24e-01 | par[  2]=SR_parm[1]
phase= 4 | nvar= 50 | iter=  0 | nll=7.96e+02 | mag=1.55e+01 | par[ 45]=selparm[11]
phase= 4 | nvar= 50 | iter= 20 | nll=7.95e+02 | mag=4.08e+02 | par[  6]=SR_parm[1]
phase= 4 | nvar= 50 | iter= 40 | nll=7.95e+02 | mag=6.13e+02 | par[  6]=SR_parm[1]
phase= 4 | nvar= 50 | iter= 60 | nll=7.95e+02 | mag=7.35e+00 | par[  6]=SR_parm[1]
phase= 4 | nvar= 50 | iter= 69 | nll=7.95e+02 | mag=3.70e-05 | par[ 45]=selparm[11]
phase= 5 | nvar= 60 | iter=  0 | nll=7.95e+02 | mag=3.70e-05 | par[ 55]=selparm[11]
Optimization completed after 31 s with final statistics:
 nll=794.959837 | mag=3.69930e-05 | par[ 55]=selparm[11]

Checking for estimated parameters on bounds... none found!
 writing big output now
Wrote bigoutput and bodywt for last_phase in REPORT_SECTION and before hessian, no benchmark or forecast
Calculating Hessian (60 variables): 20%, 40%, 60%, 80%, 100% done! (31 s)
Differentiating 8 derived quantities: do benchmark and forecast if requested in sdphase
1, 2, 3, 4, 5, 6, 7, 8 done! (2 m 11 s)
Inverting Hessian (60 variables): 20%, 40%, 60%, 80%, 100% done! (0.006 s)
Starting standard error calculations...  done! (0.42 s)

In final section
Finish time: Thu May 12 12:34:05 2022
Elapsed time: 0 hours, 3 minutes, 15 seconds.
 Iterations: 498 -log(L): 794.96
Final gradient: 3.6993e-05

 finished COVAR.SSO
 finished forecast
 finished StdDev quantities
 mceval counter: 0
 finished posteriors
 finished SS_summary.sso
 writing big output now
 finished report.sso
Creating bootstrap files: 1 files finished
 Write new starter file
 Write new forecast file
 Write new control file
dynamic Bzero in FINAL_SECTION:  finished
setup_benchmark:
SPR_profile:  finished
Global_MSY:  finished

!!  Run has completed  !!            See warning.sso for N warnings: 4

Finished running model 'ss' after 3 m 17 s.
```

**Comment by iantaylor-NOAA** [comment]
I just looked through all the changes. This all seems great to me. Moving more into echoinput.sso ensures that nothing is lost and the console output is easier to read with less info. 
@k-doering-NOAA, it looks like the output in your example was run with rundetail = 0. Should we also consider a cleanup of output for rundetail = 1 or 2? Sorry if I'm failing to see edits to those outputs in the differences shown in github--I haven't downloaded the branch to try it out myself.

**Comment by k-doering-NOAA** [comment]
> I just looked through all the changes. This all seems great to me. Moving more into echoinput.sso ensures that nothing is lost and the console output is easier to read with less info. @k-doering-NOAA, it looks like the output in your example was run with rundetail = 0. Should we also consider a cleanup of output for rundetail = 1 or 2? Sorry if I'm failing to see edits to those outputs in the differences shown in github--I haven't downloaded the branch to try it out myself.

That was what I was thinking, too, more details in echoinput rather than the console.

And you are right about the run detail setting, thanks @iantaylor-NOAA! I'll try it with 1 and 2 as well. Any idea what most people would use as a default setting?

**Comment by Rick-Methot-NOAA** [comment]
My logic similar to Ian's.  
I'm sensing good energy on this topic and will let you run with it.  What's in echoinput today is a hodgepodge of things I needed while developing.  All those IF - write statements slow SS3 down to some degree.

**Comment by Rick-Methot-NOAA** [comment]
Johnoel got me set up with ADMB 13.0, so I can test now also.  He is open to tweaks on the ADMB side if you see something in the console output that could be even better.  make_ss_safe.bat has a few more explicit references now.

**Comment by Rick-Methot-NOAA** [comment]
correct that significant output occurs in get_posteriors even in a MLE run.  Thanks for checking.

**Comment by kellijohnson-NOAA** [comment]
@Rick-Methot-NOAA I noticed that this pull request is still marked as a draft request. I wanted to make sure that it is still needed/wanted before I tried to fix the merge conflicts.

**Comment by k-doering-NOAA** [comment]
@kellijohnson-NOAA Yes, we were just waiting until using admb 13 until merging in! So I think fixing merge conflicts would be helpful, but it could also be done later.

**Comment by Rick-Methot-NOAA** [comment]
I have changes in these files queued in the PR for D-M and Tweedie.  Let me merge that one first.  Great if both of you could take a look at it first.  Ian reviewed it last week.

**Comment by Rick-Methot-NOAA** [comment]
in progress.  @nschindler-noaa is resolving merge conflicts

**Comment by nschindler-noaa** [comment]
All conflicts resolved. Just a matter of beginning/ending reading file comments and new message system (and new messages at end of run)).

**Comment by Rick-Methot-NOAA** [comment]
Excellent.  Thanks Neal.

**Comment by nschindler-noaa** [comment]
Looks like there are more warnings. Is this due to ADMB 13?

**Comment by Rick-Methot-NOAA** [comment]
Ian noticed more warnings also, but we are not using ADMB 13 yet.  Warnings seen benign, so we will adjust the expected warning count after switching to ADMB 13.  @iantaylor-NOAA

File: SS_benchfore.tpl
Comment (kellijohnson-NOAA) [inline SS_benchfore.tpl:901]: Should this line have brackets?

File: SS_benchfore.tpl
Comment (kellijohnson-NOAA) [inline SS_benchfore.tpl:1066]: Is there a difference between "Finished" and "Calculated"? If no, then I suggest changing "Finished" to "Calculated".

File: SS_benchfore.tpl
Comment (kellijohnson-NOAA) [inline SS_benchfore.tpl:1671]: Needs curly braces.

File: SS_benchfore.tpl
Comment (kellijohnson-NOAA) [inline SS_benchfore.tpl:1678]: Needs curly braces.

File: SS_global.tpl
Comment (kellijohnson-NOAA) [inline SS_global.tpl:932]: Do we want both cout and echoinput here like it is below?

File: SS_global.tpl
Comment (kellijohnson-NOAA) [inline SS_global.tpl:1064]: Curly braces should be on a new line; but, I am not sure if the automatic formatting does this?

File: SS_global.tpl
Comment (kellijohnson-NOAA) [inline SS_global.tpl:1082]: Needs curly braces.

File: SS_benchfore.tpl
Comment (Rick-Methot-NOAA) [reply SS_benchfore.tpl:1678]: Kelli,
The style guide we are using does not require {} for single statement conditionals

File: SS_global.tpl
Comment (kellijohnson-NOAA) [inline SS_global.tpl:1120]: Space after files.

File: SS_benchfore.tpl
Comment (Rick-Methot-NOAA) [reply SS_benchfore.tpl:1066]: correct.  Please change it to Calculated

File: SS_benchfore.tpl
Comment (kellijohnson-NOAA) [reply SS_benchfore.tpl:1678]: Okay, thanks for the clarification Rick, I will remove my comments below and maybe add some language to the document Neal started because right now it says  "Curly braces should always be used."

File: SS_global.tpl
Comment (kellijohnson-NOAA) [inline SS_global.tpl:1124]: I suggest a more verbose statement such as 
```
"Finished writing *.ss_new output files."
```

File: SS_objfunc.tpl
Comment (kellijohnson-NOAA) [inline SS_objfunc.tpl:409]: Not sure that it matters but there are two spaces after "obj_fun".

File: SS_param.tpl
Comment (kellijohnson-NOAA) [inline SS_param.tpl:21]: Why are these statements on line 20 and 21 different? If you keep line 21 as is, I think there needs to be a space after "parameters" before the l-dots.

File: SS_popdyn.tpl
Comment (kellijohnson-NOAA) [inline SS_popdyn.tpl:110]: There is some inconsistency throughout the files in this pull request regarding the spacing, e.g.,
```
echoinput << " text" << endl;
```
versus
```
echoinput << "text" << endl;
```
There could be a reason for this; but, here MGsetup and growth1 have a pre-space but natmort doesn't.

File: SS_popdyn.tpl
Comment (kellijohnson-NOAA) [inline SS_popdyn.tpl:331]: I would think that here and below on line 434 that if you have a statement that starts with "Begin" you would want a subsequent line in the code that has "Finished"?

File: SS_prelim.tpl
Comment (kellijohnson-NOAA) [inline SS_prelim.tpl:1244]: Earlier you used "MGsetup" to better match the code base.

File: SS_prelim.tpl
Comment (kellijohnson-NOAA) [inline SS_prelim.tpl:1393]: Change "done" to "Finished biological calculations"? Not sure on the exact text that would be needed but done is not very descript.

File: SS_popdyn.tpl
Comment (Rick-Methot-NOAA) [reply SS_popdyn.tpl:331]: I agree that would be a better style, but these statements are only in the code to assist in debugging when a weird result is occurring.  The delineation is implied when the next section begins.

File: SS_benchfore.tpl
Comment (nschindler-noaa) [reply SS_benchfore.tpl:901]: I don't see the "cout <<" statement in main. In this case, there should be brackets since there are two statements dependent on the condition.

Comment (nschindler-noaa) [reply SS_benchfore.tpl:901]: Sorry, in main it is "cout <<". In console cleanup, it is "echoinput <<".

Comment (nschindler-noaa) [reply SS_benchfore.tpl:901]: Single commands only, so brackets not required.

File: SS_benchfore.tpl
Comment (nschindler-noaa) [reply SS_benchfore.tpl:1671]: I was mis-reading these. sorry. It is a single statement and brackets/braces are not required. However, I do like to use them.

**Comment by nschindler-noaa** [comment]
@Rick-Methot-NOAA , is it okay to merge even with more warnings?

**Comment by Rick-Methot-NOAA** [comment]
Can you find and send me the warnings file so we can review it before merging with the increased N warnings

**Comment by kellijohnson-NOAA** [comment]
@Rick-Methot-NOAA I resolved all of the comments that I made with the last 7 commits. I think this is ready for your review because if my memory serves me correctly the first failing job will potentially change with the use of ADMB 13.0 and or the number of warnings expected needs to be changed.

**Comment by k-doering-NOAA** [comment]
@kellijohnson-NOAA that is correct re the warnings job

**Review by Rick-Methot-NOAA — APPROVED** [review]
Thanks Kathryn, Neal and Kelli.  Looks great.  Approving now.

**Pull request merged**
